php/libraries/Database.class.inc

```diff
@@ -327,6 +375,13 @@ class Database
         $exec_params = array();
         $prepQ      .= $this->_implodeAsPrepared(",", $set, $exec_params);
 
+        if ($onDuplicateUpdate) {
+            $prepQ .= ' ON DUPLICATE KEY UPDATE ';
+            $prepQ .= $this->_implodeAsPrepared(",", $set, $exec_params);
```

Comment:
from looking at line 382, should this line have a space after the comma in quotes? also, could we keep consistent with double and single quotes or is there a reason for the use of both here?

Reply:
No, no space is necessary after the comma here. The `_implodeAsPrepared` function takes care of that. And I left it as a double quotes because the line is identical to line `380`.  If anything, in the context of this class, the single quotes on lines `379`, `381` and `382` look out of place. But I really don't mind either way tbh.